idea/src/org/jetbrains/kotlin/idea/quickfix/ReplaceCallFixUtils.kt

```diff
@@ -5,26 +5,13 @@
 
 package org.jetbrains.kotlin.idea.quickfix
 
-import com.intellij.openapi.editor.Editor
-import com.intellij.openapi.project.Project
-import com.intellij.psi.PsiDocumentManager
-import com.intellij.psi.PsiElement
 import org.jetbrains.kotlin.idea.caches.resolve.analyze
-import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.KtBinaryExpression
 import org.jetbrains.kotlin.psi.KtExpression
 import org.jetbrains.kotlin.psi.KtProperty
-import org.jetbrains.kotlin.psi.psiUtil.endOffset
-import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
 import org.jetbrains.kotlin.resolve.BindingContext
 import org.jetbrains.kotlin.resolve.calls.callUtil.getType
 
-fun KtExpression.elvisOrEmpty(notNullNeeded: Boolean): String {
```

Comment:
This and below utils are used by `ReplaceInfixOrOperatorCallFix`, so better to be here as `Utils`?

Reply:
We cannot keep this function and `moveCaretToEnd` here in the `idea` module. They have to be moved to `idea-frontend-independent` since `ReplaceCallFix` is there as well. The only function left that is specific FE1.0 is `KtExpression.shouldHaveNotNullType()` (which I could actually just move to `ReplaceCallFix.kt`...)

(Not sure if I understood your comment correctly...)

Reply:
Ok, I will at least move the 2 functions into a `ReplaceCallFixUtils.kt` file in `idea-frontend-independent`.

Reply:
Actually, can't do that... Can't have 2 classes with the same name (`ReplaceCallFixUtilsKt`) in different modules/JARs, it can result in `NoSuchMethodError`s. I'll keep it here for now, eventually we might remove `ReplaceCallFixUtils.kt` from the `idea` module.

Reply:
Make sense. Very minor, but maybe leave a note/TODO, like one day, infix-or-operator fix is migrated as well, then we can make these utils. :)